client/src/views/HomePage.vue

```diff
@@ -193,13 +194,28 @@ export default {
     Api().get('uploads/filedata/all').then((response) => {
       this.files = response.data;
     });
+    if (!this.$store.state.user) {
+      Api().get('users/self').then((response) => {
+        this.$store.dispatch('setUser', response.data);
+      }).catch((error) => {
+        console.log(error);
+      });
+    }
   },
   methods: {
+    loginWithGoogle () {
+      window.location.href = 'http://localhost:3000/auth/google';
+    },
     logout () {
       this.$store.dispatch('setToken', null);
       this.$store.dispatch('setUser', null);
-      this.$router.push({
-        name: 'home'
+      Api().post('auth/logout').then((response) => {
+        console.log(response);
```

Comment:
Was this `console.log` intentionally left included? I have no problem with it if it is, but thought I'd point it out.

Reply:
Nope, removed it